packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,217 +73,375 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
+            &[arrow],
+        )?;
+        let batches_prototype = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/batch.js",
+            &[arrow, hash_util],
+        )?;
+
+        let ctx_import = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/context.js",
+            &[hash_util],
+        )?;
+        let ctx_import: v8::Local<'_, v8::Array> = ctx_import.try_into().map_err(|e| {
+            JSRunnerError::FileImport(
+                "./src/worker/runner/javascript/context.js".to_string(),
+                format!("Couldn't get array (of functions) from 'context.js': {e}"),
             )
         })?;
-        let experiment_ctx_prototype = ctx_import.get(0)?;
-        let sim_init_ctx_prototype = ctx_import.get(1)?;
-        let gen_ctx = ctx_import.get(2)?;
-
-        let gen_state = import_file(mv8, "./src/worker/runner/javascript/state.js", vec![
-            &hash_util,
-        ])?;
-        let fns = import_file(mv8, "./src/worker/runner/javascript/runner.js", vec![
-            &arrow,
-            &batches_prototype,
-            &experiment_ctx_prototype,
-            &sim_init_ctx_prototype,
-            &gen_ctx,
-            &gen_state,
-        ])?;
-        let fns = fns.as_array().ok_or_else(|| {
-            Error::FileImport(
+        let experiment_ctx_prototype = ctx_import.get_index(scope, 0).ok_or_else(|| {
+            JSRunnerError::V8("Could not get index 0 from context.js array".to_string())
+        })?;
+        let sim_init_ctx_prototype = ctx_import.get_index(scope, 1).ok_or_else(|| {
+            JSRunnerError::V8("Could not get index 1 from context.js array".to_string())
+        })?;
+        let gen_ctx = ctx_import.get_index(scope, 2).ok_or_else(|| {
+            JSRunnerError::V8("Could not get index 2 from context.js array".to_string())
+        })?;
+
+        let gen_state = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/state.js",
+            &[hash_util],
+        )?;
+        let fns = import_file(
+            scope,
+            context,
+            "./src/worker/runner/javascript/runner.js",
+            &[
+                arrow,
+                batches_prototype,
+                experiment_ctx_prototype,
+                sim_init_ctx_prototype,
+                gen_ctx,
+                gen_state,
+            ],
+        )?;
+        let fns: v8::Local<'_, v8::Array> = fns.try_into().map_err(|e| {
+            JSRunnerError::FileImport(
                 "./src/worker/runner/javascript/runner.js".into(),
-                "Couldn't get array (of functions) from 'runner.js'".into(),
+                format!("Couldn't get array (of functions) from 'runner.js': {e}"),
             )
         })?;
-        Ok(Self {
+
+        let start_experiment: v8::Local<'_, v8::Function> = fns
```

Comment:
There is a lot code duplication, maybe move this into a function?

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/02af15786b564ada4f9992194281ff0a3281afb9

Reply:
This was not addressed

Reply:
Confused it with https://github.com/hashintel/hash/pull/414#discussion_r833317267
Actually addressed in https://github.com/hashintel/hash/pull/414/commits/b0071adf00a0ec4f9dd928a5a52a9cc316a4f7c6

Reply:
No worries 🙂